cli/src/main/java/org/eclipse/hono/cli/client/ApplicationClient.java

```diff
@@ -0,0 +1,351 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli.client;
+
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+
+import org.eclipse.hono.cli.AbstractCliClient;
+import org.eclipse.hono.cli.adapter.CommandAndControl;
+import org.eclipse.hono.cli.adapter.TelemetryAndEvent;
+import org.eclipse.hono.cli.application.Commander;
+import org.eclipse.hono.cli.application.Receiver;
+import org.eclipse.hono.cli.shell.InputReader;
+import org.eclipse.hono.cli.shell.ShellHelper;
+import org.eclipse.hono.client.ApplicationClientFactory;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.eclipse.hono.util.Strings;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.shell.standard.ShellCommandGroup;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+import org.springframework.shell.standard.ShellOption;
+import org.springframework.stereotype.Component;
+import org.eclipse.hono.cli.client.ClientConfig.MessageTypeProvider;
+
+import io.vertx.core.Vertx;
+import sun.misc.Signal;
+import sun.misc.SignalHandler;
```

Comment:
are you sure that these are needed?

Reply:
Yep, It's to catch the interrupt signal from console

Reply:
my point is more that it is usually considered bad practice to depend on classes from the JDK implementation, in this case from `sun.*`. Is there an alternative way of achieving what you are trying to do without depending on these classes?

Reply:
Actually i tried and this way doesn't catch the sigkill from terminal. I tried also with `addShutdownHook` but doesn't work. I'm not sure how to handle it.
Actually i removed that and the side effect is that with ctrl+c you end the whole execution. Any suggestion? Also to run it from command line do i need to use the "*-exec.jar" right? How i can build it? Running maven i compile only the "*.jar" and the "*-source.jar"